MatterControlLib/ActionBar/TemperatureWidgetBed.cs

```diff
@@ -147,14 +149,31 @@ private GuiWidget GetPopupContent(ThemeConfig menuTheme)
 			{
 				if (e is StringEventArgs stringEvent)
 				{
-					var temp = printer.Settings.GetValue<double>(SettingsKey.bed_temperature);
-					valueField.Value = temp;
-					graph.GoalValue = temp;
-					settingsRow.UpdateStyle();
-					if (stringEvent.Data == SettingsKey.bed_temperature
-						&& heatToggle.Checked)
+					string settingsKey = stringEvent.Data;
+					if (this.allUiFields.TryGetValue(settingsKey, out UIField uifield))
 					{
-						SetTargetTemperature(temp);
+						string currentValue = settingsContext.GetValue(settingsKey);
+						if (uifield.Value != currentValue)
+						{
+							uifield.SetValue(
+								currentValue,
+								userInitiated: false);
+						}
+					}
+
+					if (stringEvent.Data == SettingsKey.bed_temperature)
+					{
+						var temp = printer.Settings.GetValue<double>(SettingsKey.bed_temperature);
+						graph.GoalValue = temp;
+
+						// TODO: Why is this only when enabled? How does it get set to 
```

Comment:
@larsbrubaker  - Would appreciate a review of these TODO concerns tomorrow

Reply:
The idea is the user can change profile temps without turning on the heaters. The heaters only turn on when the switch is thrown and then, in that condition, changing the temp also changes the printer temp.

I'll leave this open and we can look at it more tomorrow.